up-l1/mqtt_5.adoc

```diff
@@ -24,165 +24,215 @@ MQTT is an OASIS standard messaging protocol for the Internet of Things (IoT). I
-| ("12", "{UAttributes.payload_format}")
-| String
+[.specitem,oft-sid="req~up-transport-mqtt5-attribute-traceparent~1",oft-needs="impl,utest"]
+--
+* *MUST* be set to the traceparent value as a string.
+--
+
+| "12"
+| "{UAttributes.payload_format}"
 a| The format for the data stored in the UMessage
 
-* MUST be set to the enum value for UAttributes.payload_format link:../up-core-api/uprotocol/v1/uattributes.proto[UPayloadFormat enum].
+[.specitem,oft-sid="req~up-transport-mqtt5-attribute-payload-format~1",oft-needs="impl,utest"]
+--
+* *MUST* be set to the enum value for UAttributes.payload_format link:../up-core-api/uprotocol/v1/uattributes.proto[UPayloadFormat enum].
+--
 
 |===
 
 All uAttributes are mapped to a MQTT header UserProperty, where the key is the UAttributes protobuf field number. The value is a string representation of the UAttributes field. Only UAttributes that are used in the message are included in the MQTT header. If a UAttributes field is not present in the header, than it is considered not used when recompiling the UAttributes.
 
-=== URI Mapping to MQTT topics
+== Payload Encoding
 
-The MQTT topic a message is published on utilizes the source and sink UUri fields.
+[.specitem,oft-sid="req~up-transport-mqtt5-payload-encoding~1",oft-needs="impl,utest"]
+--
+* The MQTT payload **MUST** be the `UMessage.payload` field, which is a byte array to reduce size.
+--
 
-`{client_identifier}/{source.authority_name}/{source.ue_id}/{source.ue_version_major}/{source.resource_id}/{sink.authority_name}/{sink.ue_id}/{sink.ue_version_major}/{sink.resource_id}`
 
-If the no sink uuri is provided, then the topic sections using sink uuri segments are removed, like so:
+== MQTT5 Topics
 
-`{client_identifier}/{source.authority_name}/{source.ue_id}/{source.ue_version_major}/{source.resource_id}`
+The MQTT topic a message is published on utilizes the source and sink UUri fields. The topic is dependent on the use case for the transport implementation that will be discussed below.
 
-The client_identifier is used to determine where messages are coming from. The valid values for the client_identifier are:
 
-- `c` : Indicates that the message originated from the cloud
-- `d` : Indicates that the message originated from a device and is remote
+=== uE-2-uE Communication
 
-The client_identifier that is used for the topic is based on the kind of MQTT uTransport client that is sending the message. An MQTT uTransport client can be one of the following:
+When a message is sent from one uE to another uE, the topic is constructed using the source and sink UUri fields. The topic is constructed as follows using the string representation of a `UUri`:
 
-- Cloud MQTT uTransport client: A client that runs in the cloud.
-- Remote MQTT uTransport client: A client that runs on a device and is connected to a cloud MQTT broker.
+[.specitem,oft-sid="req~up-transport-mqtt5-ue2ue-topic~1",oft-needs="impl,utest"]
+--
+`{UAttributes.source}/{UAttributes.sink}/`
```

Comment:
what is the purpose of the trailing `/` character?

Reply:
also, what you have specified means that the topic always starts with a `/` character, which is unusual from my experience

Reply:
Removed the trailing '/'. Topic won't start with '/' as every message will always have a source but not always a sink.